cfn-resources/org-invitation/cmd/resource/resource.go

```diff
@@ -0,0 +1,241 @@
+			cloudformation.HandlerErrorCodeInvalidRequest), nil
+	}
+
+	invitationReq := &mongodbatlas.Invitation{
+		TeamIDs: currentModel.TeamIds,
+		Roles:   currentModel.Roles,
+	}
+
+	_, res, err := client.Organizations.UpdateInvitationByID(context.Background(), *currentModel.OrgId, *currentModel.Id, invitationReq)
+	if err != nil {
+		_, _ = log.Warnf("Update - error: %+v", err)
+		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
+	}
+	_, _ = log.Debugf("%s invitation updated", *currentModel.Id)
+
+	// Response
+	return Read(req, prevModel, currentModel)
+}
+
+func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	_, _ = log.Debugf("Delete() currentModel:%+v", currentModel)
+
+	// Validation
+	modelValidation := validateModel(DeleteRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		_, _ = log.Warnf("Delete - error: %+v", err)
+		return handler.ProgressEvent{
+			HandlerErrorCode: cloudformation.HandlerErrorCodeInvalidRequest,
+			Message:          err.Error(),
+			OperationStatus:  handler.Failed,
+		}, nil
+	}
+
+	res, err := client.Organizations.DeleteInvitation(context.Background(), *currentModel.OrgId, *currentModel.Id)
+	if err != nil {
+		_, _ = log.Warnf("Delete - error: %+v", err)
+		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
+	}
+	_, _ = log.Debugf("deleted invitation with Id :%s", *currentModel.Id)
+
+	// Response
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		ResourceModel:   nil,
+	}, nil
+}
+
+func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	_, _ = log.Debugf("List() currentModel:%+v", currentModel)
+
+	// Validation
+	modelValidation := validateModel(ListRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		_, _ = log.Warnf("List - error: %+v", err)
+		return progressevents.GetFailedEventByCode(fmt.Sprintf("Failed to Create Client : %s", err.Error()),
+			cloudformation.HandlerErrorCodeInvalidRequest), nil
+	}
+
+	listOptions := &mongodbatlas.InvitationOptions{
+		Username: *currentModel.Username,
+	}
+	invitations, res, err := client.Organizations.Invitations(context.Background(), *currentModel.OrgId, listOptions)
+	if err != nil {
+		_, _ = log.Warnf("List - error: %+v", err)
+		return progressevents.GetFailedEventByResponse(err.Error(), res.Response), nil
+	}
+
+	var invites []interface{}
+	// iterate invites
+	for i := range invitations {
+		invite := &Model{}
```

Comment:
```suggestion
```

Reply:
We have to pass a param of type *Model hence this is required